recipes/versionparsing.jl/scripts/activate.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+
+# build_base.sh will insert lines defining name, uuid, version, and git_url
+
+pkgsrc="${CONDA_PREFIX}/share/julia/clones/$name.jl"
+
+julia -e"
+using Pkg, UUIDs
+name = \"${name}\"
+uuid = UUID(\"${uuid}\")
+version_str = \"${version}\"
+version = v\"${version}\"
+git_rev = \"v${version}\"
+pkgsrc = \"${pkgsrc}\"
+local_git = joinpath(pkgsrc, \".git\")
+local_git_temp = joinpath(pkgsrc, \"git\")
+remote_git = \"${git_url}\"
+if !haskey(Pkg.dependencies(), uuid)
+    if !isdir(local_git) && isdir(local_git_temp)
+        # Move git to .git
+        mv(local_git_temp, local_git)
+    end
+    if isdir(local_git)
+        # Add from local, packaged git repo
+        spec = PackageSpec(
+            name=name,
+            uuid=uuid,
+            path=pkgsrc,
+            rev=git_rev,
+        )
+        Pkg.add(spec)
```

Comment:
Can you describe what this does in the file system? Create a Manifest.toml?

Reply:
`Pkg.add` will:
* Add an entry to the `Project.toml` of the default environment
* Add entries to `Manifest.toml` of the default environment for the package and its dependencies (this package has no dependencies except the standard library)
* Create a read-only copy in `$PREFIX/share/julia/packages/<hash>` where `<hash>` is a tree hash combining the git hashes and the package's UUID.

Reply:
Sorry. I meant to reply, but was busy with research. One of my main concerns is that having 100 packages each call a julia process is going to be extremely slow when activating an environment. I'd like to fix this and get this right the first time so we don't have to fix the 100 packages after we make them.

I was thinking maybe we could generate the entries to Project.toml above in the conda-build process, store them somewhere and append them using a simple bash script to the Project.toml.
This bash script can be in `julia-conda-packaging` and when the julia environment it will look into at all the parts of Project.toml we stored and concatenate them to a single file.
I'm not sure how feasible this is or how long it would take. What do you think?

Reply:
I think we should actually focus on the Manifest.toml rather than the Project.toml. The Project.toml usually does not list all the dependencies, and it does not need to list them. It only needs to list the packages that we want to directly interact with.

The Manifest.toml does list all of the dependencies as well as other critical information such as where the package was installed from.

The current build script actually does generate the Project.toml and Manifest.toml entries that we need. It's just a matter of figuring out where to package and store the files created. In the case of PySR, we did package and actually use them.

The entry for VersionParsing.jl in the Manifest.toml looks like this.

```
[[deps.VersionParsing]]
git-tree-sha1 = "58d6e80b4ee071f5efd07fda82cb9fbe17200868"
repo-rev = "v1.3.0"
repo-url = "../../clones/VersionParsing.jl"
uuid = "81def892-9a0e-5fdd-b105-ffc91e053289"
version = "1.3.0"
```

Here the Manifest.toml is located in `share/julia/environments/CONDA_PREFIX` and the packaged repository is located in `share/julia/clones/VersionParsing.jl`.

We can programatically list the contents of Manifest.toml via `Pkg.dependencies()`:

```julia
julia> Pkg.dependencies()
Dict{Base.UUID, Pkg.API.PackageInfo} with 1 entry:
  UUID("81def892-9a0e-5fdd-b105-ffc91e053289") => PackageInfo("VersionParsing", v"1.3.0", "58d6e80b4ee071f5efd07fda82cb9fbe17200868", true, false, false, true, false, "v1.3.0", "/tmp/tmp.HivkvrWrLq/shar…
```

Note that the activate script contains this line, `if !haskey(Pkg.dependencies(), uuid)`. Essentially the activate script checks if the package is in the _Manifest.toml_ rather than the Project.toml. If it's already there, it takes no further action.

I think we are going to still need the contents of the activate script. However, I think we only really need to run it once for a conda environment. Preferably we would build a Manifest.toml that only conda is managing, and then perhaps all we have to do is diff-patch that to the user's Manifest.toml.

Reply:
> This bash script can be in `julia-conda-packaging` and when the julia environment it will look into at all the parts of Project.toml we stored and concatenate them to a single file.

This has been implemented here:
https://github.com/conda-forge/staged-recipes/pull/20981

It's more complicated than concatenating Project.toml files into a single file. We also need to build a Manifest.toml and point them at our packages, so we still need a single Julia process.